src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,81 @@
-export const Pagination = () => {};
+import { FC } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+import { Props } from '../../types';
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const amountOfPages = getNumbers(1, Math.ceil(total / perPage));
+
+  const handlePageChange = (event: React.MouseEvent<HTMLElement>) => {
+    const element = event.target as HTMLElement;
+    const page = element.textContent;
+
+    onPageChange(Number(page));
```

Comment:
The `handlePageChange` function assumes that the clicked element's text content is always a valid page number. Consider adding a check to ensure that `page` is a valid number before calling `onPageChange`.